src/benchmarklib/abstract_table_generator.cpp

```diff
@@ -401,9 +401,23 @@ void AbstractTableGenerator::_create_table_indexes(
   std::cout << "- Creating table indexes done (" << format_duration(metrics.table_index_duration) << ")" << std::endl;
 }
 
-AbstractTableGenerator::IndexesByTable AbstractTableGenerator::_indexes_by_table() const {
-  // Indexes can be specified in a derived concrete class by overriding this function.
-  return {};
+AbstractTableGenerator::IndexesByTable AbstractTableGenerator::_indexes_by_table(std::unordered_map<std::string, BenchmarkTableInfo>& table_info_by_name) const {
+  auto indexes_by_table = std::map<std::string, std::vector<std::vector<std::string>>>{};
+
+  for (const auto& table_info_by_name_pair : table_info_by_name) {
+    const auto& table_name = table_info_by_name_pair.first;
+    const auto& table_info = table_info_by_name_pair.second;
+    const auto table = table_info.table;
+    const auto& column_names = table->column_names();
+
+    auto column_vectors = std::vector<std::vector<std::string>>{};
+    for (const auto& column_name : column_names) {
+      column_vectors.emplace_back(std::vector<std::string>{column_name});
+    }
+    indexes_by_table[table_name] = column_vectors;
```

Comment:
Hmm? Do we add an index on every column?

Reply:
Yepp.
Of course, this needs to be changed because it takes way too long.
But currently, the TPC-H is the only benchmark that uses indexes, so I needed a solution to test the new IndexScan in the benchmark where we hope to archive improvements.

Reply:
This is rather WIP, just to see if the index scan rule would select an index scan when all columns are indexed, right?

Reply:
This is right.